### PR TITLE
STITCH-1720 - Implement Server (Node.js) SDK

### PR DESCRIPTION
**Pull request by edaniels**

### CODE REVIEW

File: packages/server/core/src/core/auth/internal/StitchAuthImpl.ts [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+
+import version from "../../internal/common/Version";
+import AuthProviderClientFactory from "../providers/internal/AuthProviderClientFactory";
+import NamedAuthProviderClientFactory from "../providers/internal/NamedAuthProviderClientFactory";
+import StitchAuth from "../StitchAuth";
+import StitchAuthListener from "../StitchAuthListener";
+import StitchUser from "../StitchUser";
+import StitchUserFactoryImpl from "./StitchUserFactoryImpl";
+
+/**
```
Comment (adamchel) [inline packages/server/core/src/core/auth/internal/StitchAuthImpl.ts]: [nit] Don't think we need any of these `Partial` things.

File: packages/server/core/src/core/auth/internal/StitchAuthImpl.ts [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+ * Partial JSDOM window.location containing only the
+ * properties and method we need
+ */
+interface PartialLocaion {
```
Comment (adamchel) [inline packages/server/core/src/core/auth/internal/StitchAuthImpl.ts]: [nit] assuming this same typo exists in browser, you should probably drive-by fix it there as well as delete here.

File: packages/server/core/src/core/auth/internal/StitchAuthImpl.ts [hunk 3]
```diff
@@ -0,0 +1,184 @@
...
+      info[DeviceFields.DEVICE_ID] = this.deviceId;
+    }
+    if (this.appInfo.localAppName !== undefined) {
+      info[DeviceFields.APP_ID] = this.appInfo.localAppName;
+    }
+    if (this.appInfo.localAppVersion !== undefined) {
+      info[DeviceFields.APP_VERSION] = this.appInfo.localAppVersion;
+    }
+
+    info[DeviceFields.PLATFORM] = "server";
```
Comment (adamchel) [inline packages/server/core/src/core/auth/internal/StitchAuthImpl.ts]: [minor] shouldn't this be `node-server` or `js-server`?

File: packages/server/sdk/README.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+    client.close();
+})
+```
+
+6. Run the app by running `node src/index.js`.
+
+### Using the SDK
+
+#### Initialize the SDK
+1. When your app or webpage is initialized, run the following code to initialize the Stitch SDK, replacing `<your-client-app-id>` with your Stitch application's client app ID:
```
Comment (adamchel) [inline packages/server/sdk/README.md]: [nit] app or webpage -> Node.js service

File: packages/server/core/src/core/auth/internal/StitchAuthImpl.ts (hunk 3 above)
Comment (edaniels) [reply packages/server/core/src/core/auth/internal/StitchAuthImpl.ts]: yes
